scripts/etc/init.d/rcS

```diff
@@ -533,66 +533,66 @@ fi
 echo "Starting install process..."
 cdebootstrap-static --arch=armhf $cdebootstrap_cmdline $release /rootfs $mirror --keyring=/usr/share/keyrings/raspbian-archive-keyring.gpg || fail
 
-# allow root login
 echo "Configuring installed system:"
-if [ ! "$disable_root" = "1" ]; then
-    echo -n "  Setting root password... "
-    echo -n root:$rootpw | chroot /rootfs /usr/sbin/chpasswd || fail
-    echo "OK"
-fi
 
-# add SSH key for root (if provided)
-if [ ! -z "$root_ssh_pubkey" ]; then
-    echo -n "  Setting root SSH key... "
-    mkdir -p /rootfs/root/.ssh && \
-        chmod 700 /rootfs/root/.ssh && \
-        echo "$root_ssh_pubkey" > /rootfs/root/.ssh/authorized_keys
-    if [ $? -eq 0 ]; then
+# configure root login
+if [ ! "$disable_root" = "1" ]; then
+    # add SSH key for root (if provided)
+    if [ ! -z "$root_ssh_pubkey" ]; then
+        echo -n "  Setting root SSH key... "
+        mkdir -p /rootfs/root/.ssh && \
+            echo "$root_ssh_pubkey" > /rootfs/root/.ssh/authorized_keys || fail
+        echo "OK"
+        echo -n "  Setting permissions on root SSH directory... "
+        chmod -R go-rwx /rootfs/root/.ssh || fail
```

Comment:
Those permissions are too open according to [OpenSSH's Guidelines](http://www.openssh.com/faq.html#3.14)